db/db_impl/db_impl.cc

```diff
@@ -3805,6 +3811,9 @@ Iterator* DBImpl::NewIterator(const ReadOptions& _read_options,
     read_options.io_activity = Env::IOActivity::kDBIterator;
   }
 
+  read_options.total_order_seek |=
+      immutable_db_options_.prefix_seek_opt_in_only;
+
```

Comment:
Is this the best (or only) place to do this? We also have `NewIterators` as well as some other features like transactions which call `NewIteratorImpl` directly

Reply:
Fixing. I'll have an assertion in DBIter::DBIter that `prefix_seek_opt_in_only` has been applied to `total_order_seek`